utils/prof_utils.py

```diff
@@ -0,0 +1,1177 @@
+
+    Returns:
+    --------
+    sn: float
+        The estimated signal to noise ratio
+    u_sn: float
+        The estimated signal to noise ratio's its uncertainty
+    flags: list
+        VERBOSE - a list of flagged data points
+    w_equiv_bins: float
+        VERBOSE - the equivalent width of the profile measured in bins
+    u_w_equiv_bins: float
+        VERBOSE - the uncertaintiy in w_equiv_bins
+    w_equiv_ms: float
+        VERBOSE - the equivalent width of the profile measured in ms
+    u_w_equiv_ms: float
+        VERBOSE - the uncertainty in w_equiv_ms
+    scattered: boolean
+        VERBOSE - when true, the profile is highly scattered
+    """
+    if prof_path is None and (prof_data is None or period is None):
+        logger.warning("Insufficient information to attain SN estimate from profile. Returning Nones")
+        return None, None
+
+    if prof_data is None:
+        _, _, _, period, _, _, _, prof_data, nbins = get_from_bestprof(prof_path)
+        nbins = float(nbins)
+        period = float(period)
+    else:
+        nbins = len(prof_data)
+
+    #centre the profile around the max
+    shift = -int(np.argmax(prof_data))+int(nbins)//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find std and check if profile is scattered
+    sigma, flags = sigmaClip(prof_data, tol=0.01, ntrials=100, alpha=alpha)
+    bot_prof_min = (max(prof_data) - min(prof_data)) * .1 + min(prof_data)
+    scattered=False
+    if (np.nanmin(flags) > bot_prof_min) or ( not np.isnan(flags).any() ):
+        logger.warning("The profile is highly scattered. S/N estimate cannot be calculated")
+        scattered=True
+        #making a new profile with the only bin being the lowest point
+        prof_min_i = np.argmin(prof_data)
+        flags = []
+        for fi, _ in enumerate(prof_data):
+            if fi == prof_min_i:
+                flags.append(prof_data[fi])
+            else:
+                flags.append(np.nan)
+        flags = np.array(flags)
+        prof_data -= min(prof_data)
+    else:
+        prof_e = 500. #this is an approximation
+        non_pulse_bins = 0
+        #work out the above parameters
+        for i, data in enumerate(prof_data):
+            if not np.isnan(flags[i]):
+                non_pulse_bins += 1
+        sigma_e = sigma / np.sqrt(2 * non_pulse_bins - 2)
+        #now calc S/N
+        sn = max(prof_data)/sigma
+        sn_e = sn * np.sqrt(prof_e/max(prof_data)**2 + (sigma_e/sigma)**2)
+
+    if scattered==False:
+        prof_dict = prof_eval_gfit(prof_data)
+        W10 = prof_dict["W10"]
+        W10_e = prof_dict["W10_e"]
+        W50 = prof_dict["W50"]
+        W50_e = prof_dict["W50_e"]
+        Wscat = prof_dict["Wscat"]
+        Wscat_e = prof_dict["Wscat_e"]
+        Weq = prof_dict["Weq"]
+        Weq_e = prof_dict["Weq_e"]
+        maxima = prof_dict["maxima"]
+        maxima_e = prof_dict["maxima_e"]
+        chisq = prof_dict["redchisq"]
+        num_gaus = prof_dict["num_gauss"]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'num_gaus' is assigned to but never used](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)